doc/release/upcoming_changes/20394.deprecation.rst

```diff
@@ -0,0 +1,4 @@
+Removed outdated tool/allocation_tracking
+-----------------------------------------
+
+The ability to track allocations is now built-in to python.
```

Comment:
```suggestion
The ability to track allocations is now built-in to python using tracemalloc.
```
Just thought to add the term to use in search engines :).

Reply:
done